### DISCUSSION TITLE
Can't seem to create a functional order form

### DISCUSSION THREAD

**Discussion by stretchtosee** [Budibase/budibase#12369] [Help] [answered]
Hi everyone! 👋 

I'm currently working my way around Budibase and I'm loving it so far! Currently, I'm trying to create an order form for internal business use where I'm using a _Relationship picker_ to select from a list of products I have saved in a Budibase database.

Unfortunately, I can't seem to design it in such a way that I'm able to select a product from the _Relationship picker_ and then type the quantity of that product (just as you would do when issuing invoices for example). Does anyone have any ideas if this is possible?

Thanks in advance! 🙏

**Comment by ConorWebb96** [comment]
Hey @stretchtosee,

Can you send some screenshots of your table's structure as well as your form setup etc. 

This should be fairly easy to accomplish, if you have setup the relationship between your tables it should be fairly straight forward. E.g. the relationship and a number field with for the product quantity. Submitting the form should be as simple as this if you have setup the form correctly.

It might be easier if you use the [form block](https://docs.budibase.com/docs/table-block) component. This should configure and setup all of your fields for you straight out of the box based on your tables structure/schema.

I hope this information helps, thanks in advance for the additional information provided.

**Comment by stretchtosee** [comment]
Hey @ConorWebb96,

Thanks for helping me out with this. 😄 

Sure, below are some screenshots of my table structure. 
I've linked the first table to the second one through the _ordered_products_ column.

![image](https://github.com/Budibase/budibase/assets/150529163/7cf06d81-9130-4b62-958c-72ea341cedbd)

![image](https://github.com/Budibase/budibase/assets/150529163/84dbbc0a-29f4-4f0a-a133-fe3a40a31824)

I don't have a functioning form so I wouldn't be able to provide a screenshot of one, I'm afraid. What I want to achieve is to have an option within the form to choose quantity for each product selected via the _ordered_products_ column so when I display it in another app, I'm able to see a summary of the order with the respective quantities. If I add a _number_ column to mark the ordered quantity, I would be restricted to have only one product per order (so in the case of order 2 of the screenshot, I'd have to submit it as 2 separate ones).

Hopefully that makes sense. Please let me know if this is possible and/or if you need further info from me.

Thanks again!

**Comment by stretchtosee** [comment]
@ConorWebb96, sorry for the chase, but did you get a chance to have a look at this, please?

I'll be here should you need any further info about what I'm trying to achieve. Thanks!

**Reply by ConorWebb96** [reply, marked as answer]
Hey @stretchtosee,

Firstly apologies for the delayed response. I've actually been looking at this in-depth. 

This wouldn't be possible currently with the Internal Budibase table. However whenever SQS comes out it may be possible then.

I then went onto try and do this with an external database `Postgres` in my case and came up against a lot of different issues. I managed to get a some want functional app which can be found [here](https://github.com/Budibase/budibase/files/13457952/Orders.zip). Inside this zip file includes the table and import orders/products scripts to create and import data into the tables. It also includes the app used for this attempt.

I was able to get it reasonably functional with a few new features e.g. `JSON/CSV`. And the useage of the grid component. However its not at the point where I'd recommend using it.

I believe a core change will need to be implemented before this will work. The dynamic fields issue linked [here](https://github.com/Budibase/budibase/issues/5658) is really the solution to this entire issue.

**TLDR**: Currently I don't believe this sort of functionality is possible with Budibase.

I hope this helps and makes sense.

**Comment by andyburgessmd** [comment]
I'd suggest a slightly table structure might help this work.
product table - holds the product details
orderrows table - has a relationship to the product and a quantity, and maybe a formula for total price of that row
order table - holds the order details, date, client etc, and a relationship to orderrows with many rows

**Reply by stretchtosee** [reply]
Thanks a bunch for your time and insights @ConorWebb96!

@andyburgessmd, thanks as well for your suggestion. I'll be sure to try it asap. 🙏

**Reply by tresorama** [reply]
I'm trying this me too (Budibase + Postgres).

### Tables
customer:
- id, name

product:
-  id, name, price

order:
- id, customer(relationship), order_items (relationship)

order_item:
- id, order (relationship), product(relationship)

But what I usually do in "coded" app is a single form that on submit "create rows in 2 tables (order, order_item)".
how can this be done in Budibase ?
I tried adding a Form but I can link the form to only one DB Table.

**Reply by mikesealey** [reply]
@tresorama We have a guide on how to do this in our documentation [here](https://docs.budibase.com/docs/save-two-linked-tables-in-one-form)

The short version is to build a form with all of the required fields and use two "Save Row" actions. If you're creating a relationship between the two rows you create, the row_id of the 1st saved row will be available as a binding for your second "Save Row" action to use in it's relationship column.

**Reply by tresorama** [reply]
Thanks @mikesealey .
I tried following the tutorial from the docs with my use case.
I stucked with the fact that inside the actions chain the only "logic flow" available is "continue if or stop", but what I need is "if" with two following branches.
I don't know the software well but it can be a suggestion for a new feature.

**Reply by mikesealey** [reply]
@tresorama It sounds like this might be the point in your application where Budibase goes from _no-code_ to _low-code_. You can switch from the Handelbars Helpers to JavaScript in the bindings drawer of your actions. With this, you can perform the logic that you're hoping to achieve. https://docs.budibase.com/docs/application-actions#show-notification

You will also be pleased to know that branching is going to be a feature in automations in [Budibase v3](https://budibase.com/lp/3/)
